### PR TITLE
Spring boot security

### PR DESCRIPTION
**Pull request by dmytro-voronin**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/service/user/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -23,7 +25,7 @@ public UserResponseDto register(UserRegistrationRequest request) throws Registra
         }
         User user = userMapper.toModel(request);
         user.setEmail(request.getEmail());
-        user.setPassword(request.getPassword());
+        user.setPassword(passwordEncoder.encode(request.getPassword()));
         return userMapper.toDto(userRepository.save(user));
```
Comment (olekskov) [inline src/main/java/mate/academy/bookstore/service/user/impl/UserServiceImpl.java:36]: I cannot see you set any role to user during registration. Thus I don't think new user may access required endpoints

**Review by olekskov — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/bookstore/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.stereotype.Component;
+
+@EnableMethodSecurity
+@Configuration
+@Component
+
+public class SecurityConfig {
+
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/config/SecurityConfig.java]:
```suggestion
public class SecurityConfig {
```

File: src/main/java/mate/academy/bookstore/model/Role.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    @Column(nullable = false, unique = true)
+    private RoleName roleName;
+
+    public enum RoleName {
+        USER,
+        MANAGER,
+        ADMIN
+    }
+}
+
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/model/Role.java]:
```suggestion
```

File: src/main/java/mate/academy/bookstore/service/role/impl/RoleServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+@RequiredArgsConstructor
+@Component
+public class RoleServiceImpl implements RoleService {
+    private final RoleRepository roleRepository;
+
+    @Override
+    public Role getRoleByRoleName(Role.RoleName roleName) {
+        return roleRepository.findRoleByRoleName(roleName).orElseThrow(() ->
+                new RuntimeException("can't find role by roleName: " + roleName));
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/service/role/impl/RoleServiceImpl.java]: EntityNotFoundException

File: src/main/resources/db/changelog/changes/03-create-roles-table.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: role_name
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+                    unique: true
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/03-create-roles-table.yaml]: make sure each file ends with exactly one empty line
image

File: src/main/resources/db/changelog/changes/06-insert-users-table.yml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+databaseChangeLog:
+  - changeSet:
+      id: insert-users
+      author: dima
+      changes:
+        - insert:
+            tableName: users
+            columns:
+              - column: {name: email, value: "admin.test@example.com"}
+              - column: {name: password, value: "$2a$10$1pEimxSACCZhIfEYS12Bze6VtLuJYWAHCZUwLb.r9P2barlIgae5a"}
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/06-insert-users-table.yml]: add comment with this password in plaintext for testing purposes

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/bookstore/controller/BookController.java [hunk 7]
```diff
@@ -30,31 +31,36 @@ public class BookController {
 
     @Operation(summary = "Get all books", description = "Get a list of all available books")
     @GetMapping
+    @PreAuthorize("hasRole('USER')")
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/controller/BookController.java:34]: It wasn't clarified in the task description, but let's allow Admins to do user roles too, for that, you need to remove @PreAuthorize("hasRole('USER')") and leave only @PreAuthorize("hasRole('ADMIN')") - others are secured in SecurityConfig

**Review by okuzan — APPROVED** [review]

**Pull request closed without merging**